zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -103,6 +109,13 @@
   // ZkEventThread. Otherwise the retry request might block the normal event processing.
   protected final ZkAsyncRetryThread _asyncCallRetryThread;
 
+  static {
```

Comment:
I don't want to change the original init way for MAX_RECONNECT_INTERVAL_MS.
Like I comment, in terms of NUM_CHILDREN_LIMIT, the purpose is for unit test. I explain it in comment: Set it here for unit test to use reflection to change value because compilers optimize constants by replacing them inline. If we don't do it like this, we would not be able to unit test getChildren() unless we create > 100K children.

Reply:
@jiajunwang No, the test will not work, because compilers optimize it inline:
```
getChildren() {
  if (stat.getNumChildren() > 100 * 1000) { }
}
```

We would like to avoid that and keep the variable. So the static block is used.

In the version 3.4.13 of zk we use, the `packetLen` also declares in this way.
```
public static final int packetLen;

  static {
    if (LOG.isDebugEnabled()) {
      LOG.debug("zookeeper.disableAutoWatchReset is " + disableAutoWatchReset);
    }

    packetLen = Integer.getInteger("jute.maxbuffer", 4194304);
  }
```

FYI, `packetLen` is removed in newer version of ZK.

Reply:
Thanks for the explanation.